packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -201,6 +201,19 @@ def update_paging_response(yaml_data: Dict[str, Any]) -> None:
 }
 
 
+def add_missing_parameter(
+    parameters: List[Dict[str, Any]],
+    idx: int,
+    copy_idx: int,
+    name: str,
+    offset: int = 0,
+):
+    if idx < 0:
```

Comment:
i really don't like playing with indices, it makes it really not readable. As a whole it's very hard to follow what this code is doing. You can try coming up with the parameters we need, then building a parameter list. But I don't want to merge code that is copying parameters and changing names and reordering indices, there really should be a much clearer way.

Reply:
Let me introduce more context here:
(1) copy is best way to when add missing parameter since we don't need to care about other properties except name
(20 With `insert` instead of `append`, we can make sure `etag` and `match_condition` appear at adjacent position.